tests/unit/bintime/test_datetime.py

```diff
@@ -0,0 +1,478 @@
+            999_999_999,
+            999_000_000,  # with 999_999_999, binary fraction rounding pushes us to MAXYEAR + 1
+            dt.timezone.utc,
+        )
+    ),
+]
+
+
+def test___various_values___hash___returns_probably_unique_int() -> None:
+    hashes = set([hash(x) for x in _VARIOUS_VALUES])
+    assert len(hashes) == len(_VARIOUS_VALUES)
+
+
+@pytest.mark.parametrize("value", _VARIOUS_VALUES)
+def test___various_values___copy___makes_copy(value: DateTime) -> None:
+    new_value = copy.copy(value)
+    assert new_value is not value
+    assert new_value == value
+
+
+@pytest.mark.parametrize("value", _VARIOUS_VALUES)
+def test___various_values___pickle_unpickle___makes_copy(value: DateTime) -> None:
+    new_value = pickle.loads(pickle.dumps(value))
+    assert new_value is not value
+    assert new_value == value
+
+
+def test___time_value___pickle___references_public_modules() -> None:
+    value = DateTime()
+    value_bytes = pickle.dumps(value)
+
+    assert b"nitypes.bintime" in value_bytes
+    assert b"nitypes.bintime._datetime" not in value_bytes
+    assert b"nitypes.bintime._time_value" not in value_bytes
+
+
+@pytest.mark.parametrize(
+    "value, expected",
+    [
+        (
+            DateTime.min,
+            "0001-01-01 00:00:00+00:00",
+        ),
+        (
+            DateTime(
+                ht.datetime(
+                    1850, 12, 25, 8, 15, 30, 123_456, 234_567_789, 345_567_890, dt.timezone.utc
+                )
+            ),
+            "1850-12-25 08:15:30.123456234567789345578196+00:00",
+        ),
+        (
+            DateTime(
+                ht.datetime(
+                    1903, 12, 31, 23, 59, 59, 123_456, 234_567_789, 345_567_890, dt.timezone.utc
+                )
+            ),
+            "1903-12-31 23:59:59.123456234567789345578196+00:00",
+        ),
+        (
+            DateTime(ht.datetime(1904, 1, 1, 0, 30, 0, 0, 0, 1_000_000, dt.timezone.utc)),
+            "1904-01-01 00:30:00.000000000000000000975781+00:00",
+        ),
+        (
+            DateTime(ht.datetime(2000, 1, 1, 0, 0, 0, 0, 0, 0, dt.timezone.utc)),
+            "2000-01-01 00:00:00+00:00",
+        ),
+        (
+            DateTime.max,
+            "9999-12-31 23:59:59.999999999999999999945789+00:00",
+        ),
+    ],
+)
+def test___various_values___str___looks_ok(value: TimeDelta, expected: str) -> None:
+    assert str(value) == expected
+
+
+@pytest.mark.parametrize(
+    "value, expected",
+    [
+        (
+            DateTime.min,
+            "nitypes.bintime.DateTime(hightime.datetime(1, 1, 1, 0, 0, tzinfo=datetime.timezone.utc))",
+        ),
+        (
+            DateTime(
+                ht.datetime(
+                    1850, 12, 25, 8, 15, 30, 123_456, 234_567_789, 345_567_890, dt.timezone.utc
+                )
+            ),
+            "nitypes.bintime.DateTime(hightime.datetime(1850, 12, 25, 8, 15, 30, 123456, 234567789, 345578196, tzinfo=datetime.timezone.utc))",
```

Comment:
Now that the class is named DateTime, it seems silly for the constructor to require you to pass in a different datetime type. I'll add another overload that takes datetime-style arguments, and change repr accordingly.